### PR TITLE
feat: changes + added documentation to features

### PR DESCRIPTION
**Pull request by cka-y**
**Summary:**

- Updated features based on the [table description](https://github.com/MobilityData/gtfs-validator/issues/1679#issuecomment-2332601261)
- Added documentation links to the HTML report for each feature

**Expected Behavior:**
- Ensures consistent naming conventions between the HTML report and the official documentation.
- Each feature listed in the report now includes a clickable link that directs users to the corresponding section in the official documentation
Screenshot 2024-09-23 at 4 26 44 PM

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Run the unit tests with `gradle test` to make sure you didn't break anything
- [ ] Add or update any needed [documentation](https://github.com/MobilityData/gtfs-validator/tree/master/docs) to the repo 
- [x] Format the title like "feat: [new feature short description]". Title must follow the Conventional Commit Specification(https://www.conventionalcommits.org/en/v1.0.0/).
- [x] Linked all relevant issues
- [x] Include screenshot(s) showing how this pull request works and fixes the issue(s)

### CODE REVIEW

File: main/src/main/java/org/mobilitydata/gtfsvalidator/report/model/FeedMetadata.java [hunk 1]
```diff
@@ -38,32 +38,36 @@ public class FeedMetadata {
...
+          new Pair<>(new FeatureMetadata("Translations", null), GtfsTranslation.FILENAME),
+          new Pair<>(new FeatureMetadata("Fares V1", "Fares"), GtfsFareAttribute.FILENAME),
+          new Pair<>(new FeatureMetadata("Fare Products", "Fares"), GtfsFareProduct.FILENAME),
+          new Pair<>(new FeatureMetadata("Fare Media", "Fares"), GtfsFareMedia.FILENAME),
+          new Pair<>(new FeatureMetadata("Zone-Based Fares", "Fares"), GtfsArea.FILENAME),
+          new Pair<>(
+              new FeatureMetadata("Fares Transfers", "Fares"), GtfsFareTransferRule.FILENAME),
+          new Pair<>(new FeatureMetadata("Time-Based Fares", "Fares"), GtfsTimeframe.FILENAME),
+          new Pair<>(
+              new FeatureMetadata("Booking Rules", "Flexible Services"), GtfsTimeframe.FILENAME),
```
Comment (jcpitre) [inline main/src/main/java/org/mobilitydata/gtfsvalidator/report/model/FeedMetadata.java:67]: Shouldn't the parameter be GtfsBookingRules.FILENAME
and GtfsLocationGroups.FILENAME for the next line?

**Comment by emmambd** [comment]
@cka-y Looks great! Super minor feedback: 

Capitalize "In-Station Traversal Time"
Capitalize "Zone-Based Demand Responsive Services" and center text
Screenshot 2024-09-24 at 2 58 46 PM

@tzujenchanmbd @Sergiodero Do you know of specific feeds in the [production list](https://docs.google.com/document/d/1QABoIeOdR6lAB7A6_X6WIbZwC7EnAIZvm_hRNwNQeoI/edit) that have Predefined Routes with Deviation or Fixed Stops Demand Responsive Service to test with? I'm having some trouble identifying them.

**Review by jcpitre — APPROVED** [review]
LGTM!

**Comment by cka-y** [comment]
✅ done @emmambd

**Comment by tzujenchanmbd** [comment]
@emmambd 
Feeds for test:
Fixed Stops - https://data.trilliumtransit.com/gtfs/islandtransit-wa-us/islandtransit-wa-us--flex-v2-TEST.zip
Predefined Routes with Deviation - https://data.trilliumtransit.com/gtfs/browncounty-mn-us/browncounty-mn-us--flex-v2.zip

**Comment by emmambd** [comment]
@tzujenchanmbd It looks like the Predefined Routes doesn't trigger for the feed you provided. I think it's because the logic for it from #1776 was: 

`Deviated Fixed Route: At least a trip in stop_times references location_id AND stop_id AND arrival_time AND departure_time`

It looks like it's either location_id or stop_id from the data. Should the rule instead be:

Two trips: 

At least one trip in stop_times references location_id AND one trip in stop_times references stop_id, arrival_time and departure_time

**Comment by tzujenchanmbd** [comment]
@emmambd The rule should still check within 1 trip instead of 2.

[](url)
Screenshot 2024-09-25 at 12 00 45 PM

It seems the trip `t_5374696_b_77497_tn_0` does include these 4 field values, but not in the same entry. Perhaps the current logic is checking the same entry?

**Comment by emmambd** [comment]
I'll add a new issue for the problem with Predefined Routes. 

@cka-y, last comment

- Update Frequencies to Frequency-Based Service: https://gtfs.org/getting_started/features/base_add-ons/#frequency-based-service based on discussion with Sergio here https://github.com/MobilityData/gtfs-validator/pull/1812#issuecomment-2374673756

**Comment by cka-y** [comment]
done [here](https://github.com/MobilityData/gtfs-validator/pull/1844/files#diff-863142965cbc810ef70cd6f71119c830620f6aac2abb96a44a8f544b9194ec5aR64) @emmambd

**Pull request merged**